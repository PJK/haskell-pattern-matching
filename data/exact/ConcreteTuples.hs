module ConcreteTuples where

quadrupleAnd :: (Bool, Bool, Bool, Bool) -> Bool
quadrupleAnd (True, True, True, True) = True
quadrupleAnd (False, _, _, _) = False
quadrupleAnd (_, False, _, _) = False
quadrupleAnd (_, _, False, _) = False
-- T, T, T, F is missing