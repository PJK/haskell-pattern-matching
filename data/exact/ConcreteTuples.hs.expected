{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            "quadrupleAnd",
            {
                "tag": "NonExhaustive",
                "contents": [
                    [
                        [
                            {
                                "tag": "TuplePattern",
                                "contents": [
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "True",
                                            []
                                        ]
                                    },
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "True",
                                            []
                                        ]
                                    },
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "True",
                                            []
                                        ]
                                    },
                                    {
                                        "tag": "ConstructorPattern",
                                        "contents": [
                                            "False",
                                            []
                                        ]
                                    }
                                ]
                            }
                        ],
                        ""
                    ]
                ]
            }
        ]
    ]
}
